src/pip/_internal/commands/inspect.py

```diff
@@ -0,0 +1,91 @@
+import json
+import sys
+from optparse import Values
+from typing import Any, Dict, List
+
+from pip._vendor.packaging.markers import default_environment
+
+from pip import __version__
+from pip._internal.cli import cmdoptions
+from pip._internal.cli.req_command import Command
+from pip._internal.cli.status_codes import SUCCESS
+from pip._internal.metadata import BaseDistribution, get_environment
+from pip._internal.utils.compat import stdlib_pkgs
+from pip._internal.utils.urls import path_to_url
+
+
+class InspectCommand(Command):
+    """
+    Inspect the content of a Python environment.
+    """
+
+    ignore_require_venv = True
+    usage = """
+      %prog [options]"""
+
+    def add_options(self) -> None:
+        self.cmd_opts.add_option(
+            "-l",
+            "--local",
+            action="store_true",
+            default=False,
+            help=(
+                "If in a virtualenv that has global access, do not list "
+                "globally-installed packages."
+            ),
+        )
+        self.cmd_opts.add_option(
+            "--user",
+            dest="user",
+            action="store_true",
+            default=False,
+            help="Only output packages installed in user-site.",
+        )
+        self.cmd_opts.add_option(cmdoptions.list_path())
+        self.parser.insert_option_group(0, self.cmd_opts)
+
+    def run(self, options: Values, args: List[str]) -> int:
+        cmdoptions.check_list_path_option(options)
+        dists = get_environment(options.path).iter_installed_distributions(
+            local_only=options.local,
+            user_only=options.user,
+            skip=set(stdlib_pkgs),
+        )
+        output = {
+            "version": "0",
+            "pip_version": __version__,
+            "installed": [self._dist_to_dict(dist) for dist in dists],
+            "environment": default_environment(),
+            # TODO tags? scheme?
```

Comment:
Let's add the scheme here. I think we can have the tags be under a separate command, like `pip inspect tags` -- since that's a fairly overwhelming list on its own.

Reply:
Actually, I wonder if we should omit `installed` from this, putting that behind a `pip inspect installed`. Not sure TBH -- I'm basically thinking aloud with a keyboard right now.

Reply:
Hmm. This command is primarily intended for tools and for these I don't think the output size will matter that much.
Reducing the number of subprocess calls will matter though, so I'm inclined to output as much information as possible by default.

Let me see how it looks like with the rich pager. If that works well, having a big list of tags at the end should not bother much.

That said I was not even sure to add tags and scheme now as I don't have very concrete use cases for these yet.

Reply:
If we conclude it is necessary to have a short output, I'd rather have a short output by default and the a `--all` options, and/or options for specific components, so the caller can select what it needs in the output.

But I'm not quite convinced it's worth the additional complexity.

Reply:
Fair enough!